dissect/target/plugins/os/windows/firewall.py

```diff
@@ -0,0 +1,329 @@
+                    key=entry.name,
+                    **data,
+                    source=f"HKLM\\{reg.path}",
+                    _target=self.target,
+                )
+
+    @export(record=WindowsFirewallLogRecord)
+    def logs(self) -> Iterator[WindowsFirewallLogRecord]:
+        """Parse Windows Firewall log files.
+
+        Currently parses ``pfirewall*`` files in ``sysvol\\Windows\\System32\\LogFiles\\Firewall\\`` only.
+        Does not yet parse dynamically set log locations e.g. ``netsh advfirewall set currentprofile logging filename``.
+
+        Resources:
+            - https://learn.microsoft.com/en-us/windows/security/operating-system-security/network-security/windows-firewall/configure-logging
+
+        Yields Windows Firewall log records with the following fields:
+
+        .. code-block:: text
+
+            ts (datetime): The timestamp of the log entry.
+            hostname (string): The target hostname.
+            domain (string): The target domain.
+            action (string): Allow or Block.
+            protocol (string): TCP, UDP or other IANA protocol value.
+            src_ip (net.ipaddress): Source IP address.
+            dst_ip (net.ipaddress): Destination IP address.
+            src_port (varint): Source port number.
+            dst_port (varint): Destination port number.
+            size (filesize): Size in bytes of the packet(s).
+            tcpflags (string): TCP header control flags.
+            tcpsyn (string): TCP sequence number.
+            tcpack (string): TCP acknowledgement number.
+            tcpwin (string): TCP window size in bytes.
+            icmptype (string): ICMP packet type.
+            icmpcode (string): ICMP packet code.
+            info (string): Additional information.
+            path (string): Direction of the traffic, either SEND, RECEIVE, FORWARD or UNKNOWN.
+            source (path): Source path of the record log line.
+        """
+
+        for path in self.log_paths:
+            try:
+                fh = path.open("rt")
+                header = list(islice(fh, 4))
+                config = LogConfig(header)
+
+            except Exception as e:
+                self.target.log.warning("Unable to read Windows Firewall log file config in %s: %s", path, e)
+                self.target.log.debug("", exc_info=e)
+                continue
+
+            if config.version != 1.5:
+                self.target.log.warning(
+                    "Unsupported Windows Firewall log file version %r in log file %s", header[0], path
+                )
+                continue
+
+            for line in fh:
+                if not (line := line.strip()):
+                    continue
+
+                entry = FieldMap(config.fields, line.split(" ", len(config.fields)))
+                entry.ts = datetime.strptime(f"{entry.date} {entry.time}", "%Y-%m-%d %H:%M:%S").replace(
+                    tzinfo=self.target.datetime.tzinfo if config.time_format == "local" else None
```

Comment:
Is `None` correct? Wouldn't you want to set UTC? We don't want any unaware datetime objects anymore.

Reply:
Better to be explicit, yes.